### DISCUSSION TITLE
The declaration mode in switch cannot be correctly judged

### DISCUSSION THREAD

**Discussion by zms9110750** [dotnet/roslyn#61709] [General]
```csharp
int[] p = null;
switch (p is [3, .. var c])
{
	case true:
		Console.WriteLine(c);//CS0165
		break;
}
```
If the if statement can pass, the true part of the switch should also pass

**Comment by RikkiGibson** [comment]
This looks like a general problem where split definite assignment state is lost, or at least isn't threaded through `true` and `false` patterns as precisely as it should be. We do compile successfully in a scenario like the following.

```cs
int[] p = null;
switch (p)
{
	case [3, .. var c]:
		Console.WriteLine(c);
		break;
}
```
